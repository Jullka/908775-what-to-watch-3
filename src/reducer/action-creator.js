import DataActionCreator from '../reducer/data/data.js';
import UserActionCreator from '../reducer/user/user.js';

export default Object.assign({}, DataActionCreator, UserActionCreator);
