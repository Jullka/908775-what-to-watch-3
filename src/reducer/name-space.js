const NameSpace = {
  DATA: `DATA`,
  USER: `USER`
};

export {NameSpace};
