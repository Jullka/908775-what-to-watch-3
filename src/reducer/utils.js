import AppActionCreator from '../reducer/app/app.js';
import UserActionCreator from '../reducer/user/user.js';
import {AppState, AuthorizationStatus, ErrorCode} from '../components/const.js';

const handleError = (err) => (dispatch) => {
  if (err.response.status === ErrorCode.UNAUTHORIZED) {
    return dispatch(UserActionCreator.setAuthorizationStatus(AuthorizationStatus.NO_AUTH));
  }

  return dispatch(AppActionCreator.changeAppState(AppState.ERROR));
};

export {handleError};
