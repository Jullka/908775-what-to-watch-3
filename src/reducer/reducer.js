import {extend} from '../utils.js';
import {Movies, movieDetails} from '../mocks/movies.js';
import {getMoviesByGenre} from '../components/utils/get-movies-by-genre.js';
import {GameScreen} from '../components/const.js';

const ALL_GENRES = `All genres`;
const SHOWN_MOVIES_NUMBER = 8;

const initialState = {
  gameScreen: GameScreen.MAIN,
  history: [],
  movieDetails,
  selectedGenre: ALL_GENRES,
  movies: Movies,
  selectedMovie: null,
  filteredMovies: [...Movies],
  shownMoviesNumber: Movies.length > SHOWN_MOVIES_NUMBER ? SHOWN_MOVIES_NUMBER : Movies.length
};

const ActionType = {
  CHANGE_GAME_SCREEN: `CHANGE_GAME_SCREEN`,
  CHANGE_GENRE: `CHANGE_GENRE`,
  GET_MOVIES_BY_GENRE: `GET_MOVIES_BY_GENRE`,
  SELECT_MOVIE: `SELECT_MOVIE`,
  SHOW_MORE_MOVIES: `SHOW_MORE_MOVIES`,
  RESET_SHOW_MORE_MOVIES: `RESET_SHOW_MORE_MOVIES`,
  ROLLBACK: `ROLLBACK`
};

const ActionCreator = {
  changeGenre: (genre) => ({
    type: ActionType.CHANGE_GENRE,
    payload: genre
  }),
  getMoviesByGenre: () => ({
    type: ActionType.GET_MOVIES_BY_GENRE
  }),
  selectMovie: (movie) => ({
    type: ActionType.SELECT_MOVIE,
    payload: movie
  }),
  showMoreMovies: () => ({
    type: ActionType.SHOW_MORE_MOVIES,
    payload: SHOWN_MOVIES_NUMBER
  }),
  resetShowMoreMovies: () => ({
    type: ActionType.RESET_SHOW_MORE_MOVIES
  }),
  changeGameScreen: (gameScreen) => ({
    type: ActionType.CHANGE_GAME_SCREEN,
    payload: gameScreen
  }),
  rollback: () => ({
    type: ActionType.ROLLBACK
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_GAME_SCREEN:
      return extend(state, {
        gameScreen: action.payload
      });
    case ActionType.CHANGE_GENRE:
      return extend(state, {
        selectedGenre: action.payload
      });

    case ActionType.GET_MOVIES_BY_GENRE:
      return extend(state, {
        filteredMovies: getMoviesByGenre(state.movies, state.selectedGenre),
      });

    case ActionType.SELECT_MOVIE:
      return extend(state, {
        selectedMovie: action.payload
      });
    case ActionType.SHOW_MORE_MOVIES:
      return extend(state, {
        shownMoviesNumber: state.shownMoviesNumber + action.payload
      });
    case ActionType.RESET_SHOW_MORE_MOVIES:
      return extend(state, {
        shownMoviesNumber: SHOWN_MOVIES_NUMBER
      });
    case ActionType.ROLLBACK:
      state.history.pop();
      state.history.pop();
      return reducer(state, state.history.pop());

    default:
      return reducer(state, {
        type: ActionType.CHANGE_GAME_SCREEN,
        payload: GameScreen.MAIN
      });
  }
};

export {reducer, ActionType, ActionCreator};
