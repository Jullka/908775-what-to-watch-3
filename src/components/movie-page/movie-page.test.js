import React from 'react';
import renderer from 'react-test-renderer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {reducer} from '../../reducer/reducer.js';
import MoviePage from './movie-page.jsx';

const movieDetails = {
  id: `022`,
  title: `Avengers: Infinity War`,
  genre: `Comedy`,
  releaseDate: 1999,
  poster: `img/we-need-to-talk-about-kevin.jpg`,
  bigPoster: `img/we-need-to-talk-about-kevin.jpg`,
  ratingScore: 7.4,
  ratingLevel: `Very good`,
  ratingCount: 311,
  runTime: 144,
  text: `Having acquired the Power Stone, one of the six Infinity Stones, from the planet Xandar, Thanos and his lieutenants—Ebony Maw, Cull Obsidian, Proxima Midnight, and Corvus Glaive—intercept the spaceship carrying the survivors of Asgard's recent destruction.[N 1] As they extract the Space Stone from the Tesseract, Thanos subdues Thor, overpowers Hulk, and kills Loki. `,
  director: `Anthony Russo, Joe Russo`,
  starring: [`Robert Downey Jr.`, `Chris Hemsworth`, `Mark Ruffalo`, `Chris Evans`, `Scarlett Johansson`],
  reviews: [
    {
      id: `1`,
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      author: `Kate Muir`,
      date: new Date(),
      rating: 9.0
    },
    {
      id: `2`,
      text: `Ut ornare lectus sit amet. Molestie at elementum eu facilisis sed odio morbi quis. Sit amet dictum sit amet justo donec. Pulvinar neque laoreet suspendisse interdum consectetur libero id faucibus nisl.`,
      author: `Bill Goodykoontz`,
      date: new Date(),
      rating: 8.0
    },
    {
      id: `3`,
      text: `Pulvinar elementum integer enim neque volutpat ac tincidunt. Parturient montes nascetur ridiculus mus mauris vitae.`,
      author: `Amanda Greever`,
      date: new Date(),
      rating: 6.7
    },
  ]
};

const store = createStore(reducer);

it(`MoviePage renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <MoviePage
            movie={movieDetails}
          />
        </Provider>,
        {
          createNodeMock: () => ({})
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
