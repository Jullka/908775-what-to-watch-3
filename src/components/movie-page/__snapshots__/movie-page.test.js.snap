// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage renders correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Terminator 2: Judgment Day"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="main.html"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            href="/auth-dev"
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Terminator 2: Judgment Day
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thrillers
            </span>
            <span
              className="movie-card__year"
            >
              1991
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Terminator 2: Judgment Day"
            height="327"
            src="img/johnny-english.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating movie-tab movie-tab--overview
        "
          >
            <div
              className="movie-rating__score"
            />
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                
              </span>
              <span
                className="movie-rating__count"
              >
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text movie-tab movie-tab--overview
        "
          >
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Steven Zaillian
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                John Travolta, Robert Duvall, Stephen Fry
                 and other
              </strong>
            </p>
          </div>
          <div
            className="movie-card__text movie-card__row movie-tab movie-tab--details
        visually-hidden"
          >
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Director
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Steven Zaillian
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Starring
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  John Travolta
                   
                  <br />
                  Robert Duvall
                   
                  <br />
                  Stephen Fry
                   
                  <br />
                </span>
              </p>
            </div>
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Run Time
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  NaN
                  h 
                  NaN
                  m
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Genre
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Thrillers
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Released
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  1991
                </span>
              </p>
            </div>
          </div>
          <div
            className="movie-card__reviews movie-card__row movie-tab movie-tab--reviews
        visually-hidden"
          >
            <div
              className="movie-card__reviews-col"
            >
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Kate Muir
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 15, 2019"
                    >
                      March 15, 2019
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  9
                </div>
              </div>
              <br />
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    Ut ornare lectus sit amet. Molestie at elementum eu facilisis sed odio morbi quis. Sit amet dictum sit amet justo donec. Pulvinar neque laoreet suspendisse interdum consectetur libero id faucibus nisl.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Bill Goodykoontz
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 25, 2019"
                    >
                      March 25, 2019
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  8
                </div>
              </div>
              <br />
              <div
                className="review"
              >
                <blockquote
                  className="review__quote"
                >
                  <p
                    className="review__text"
                  >
                    Pulvinar elementum integer enim neque volutpat ac tincidunt. Parturient montes nascetur ridiculus mus mauris vitae.
                  </p>
                  <footer
                    className="review__details"
                  >
                    <cite
                      className="review__author"
                    >
                      Amanda Greever
                    </cite>
                    <time
                      className="review__date"
                      dateTime="March 15, 2018"
                    >
                      March 15, 2018
                    </time>
                  </footer>
                </blockquote>
                <div
                  className="review__rating"
                >
                  6.7
                </div>
              </div>
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <video
            height="175"
            width="250"
          />
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <video
            height="175"
            width="250"
          />
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <video
            height="175"
            width="250"
          />
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <video
            height="175"
            width="250"
          />
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <video
            height="175"
            width="250"
          />
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <video
            height="175"
            width="250"
          />
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <video
            height="175"
            width="250"
          />
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <video
            height="175"
            width="250"
          />
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Johnny English
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
